app.py

```diff
@@ -181,27 +179,40 @@ def _do_analyze_build(
         _METRIC_IMAGES_PUSHED_REGISTRY.inc()
         _LOGGER.info("Successfully pushed image to %r", output_reference)
 
-    build_detail = {"base_image": base_input_reference, "output_image": output_reference}
-    build_detail.update(build_log_reference)
-
-    analysis_ids = build_analysis(
-        build_detail=build_detail,
+    analysis_response = build_analysis(
+        base_image=base_input_reference,
+        output_image=output_reference,
+        build_log=build_log_reference,
         environment_type=environment_type,
         registry_user=dst_registry_user,
         registry_password=dst_registry_password,
         registry_verify_tls=dst_verify_tls,
         nowait=True,
     )
 
+    if analysis_response.base_image_analysis.analysis_id:
+        _METRIC_IMAGES_SUBMITTED.inc()
+    if analysis_response.build_log_analysis.analysis_id:
+        _METRIC_BUILD_LOGS_SUBMITTED.inc()
+    if analysis_response.output_image_analysis.analysis_id:
+        _METRIC_IMAGES_SUBMITTED.inc()
+
+    try:
+        _LOGGER.info(f"Submitting metrics to Prometheus pushgateway {_THOTH_METRICS_PUSHGATEWAY_URL}")
+        push_to_gateway(_THOTH_METRICS_PUSHGATEWAY_URL, job="build-watcher", registry=prometheus_registry)
+    except Exception as e:
+        _LOGGER.exception(f"An error occurred pushing the metrics: {str(e)}")
+
     _LOGGER.info(
-        "Successfully submitted %s,%s,%s to Thoth for analysis",
+        "Successfully submitted %s,%s,%s to Thoth for analysis; analysis ids respectively: %s, %s, %s",
```

Comment:
I like %r more (as it can uncover possible white-space issues), but its just my personal preference :)